spec/lib/evss/vso_search/service_spec.rb

```diff
@@ -26,5 +26,11 @@ def it_handles_errors(method, form = nil, form_id = nil)
     it 'handles errors' do
       it_handles_errors(:get_current_info, response.body)
     end
+
+    it 'sets the correct base headers' do
+      headers = JSON.parse(File.read('spec/fixtures/evss_vso_search/service_headers.json'))
+      allow(service).to receive(:perform).with(:post, 'getCurrentInfo', '', headers).and_return(response)
+      service.send(*[:get_current_info, response.body].compact)
```

Comment:
Theres no actual test here we should probably expect something no?

Reply:
@ATeal  thee allow to receive is the test. if you remove the content type header from or change the empty string to nil in the implementation code then this test will fail. 
ex.
`       #<EVSS::VsoSearch::Service:0x00007fc909801108 @user=#<User:0x00007fc9099683c0 @uuid="b2fab2b5-6af0-45e1-a9e2-394347af91ef", @last_signed_in=2017-12-07 00:55:09 UTC, @mhv_last_signed_in=nil, @persisted=true, @identity=#<UserIdentity:0x00007fc909808ed0 @uuid="b2fab2b5-6af0-45e1-a9e2-394347af91ef", @email="abraham.lincoln@vets.gov", @first_name="WESLEY", @middle_name=nil, @last_name="FORD", @gender="M", @birth_date="1809-02-12", @zip="17325", @ssn="796043735", @loa={:current=>3, :highest=>3}, @multifactor=false, @authn_context="http://idmanagement.gov/ns/assurance/loa/3/vets", @mhv_icn=nil, @mhv_correlation_id=nil, @mhv_account_type=nil, @dslogon_edipi=nil, @sign_in={:service_name=>"idme"}, @persisted=true, @validation_context=nil, @errors=#<ActiveModel::Errors:0x00007fc909802e18 @base=#<UserIdentity:0x00007fc909808ed0 ...>, @messages={}, @details={}>>, @validation_context=nil, @errors=#<ActiveModel::Errors:0x00007fc9098680b0 @base=#<User:0x00007fc9099683c0 ...>, @messages={}, @details={}>, @mvi=#<Mvi:0x00007fc909800fa0 @uuid=nil, @response=nil, @persisted=false, @user=#<User:0x00007fc9099683c0 ...>, @mvi_response=#<MVI::Responses::FindProfileResponse:0x00007fc90986bd78 @status="OK", @profile=#<MVI::Models::MviProfile:0x00007fc90988a728 @given_names=["Melvina", "Monte"], @family_name="Weissnat", @suffix="III", @gender="M", @birth_date="1986-05-06", @ssn="724572799", @address=#<MVI::Models::MviProfileAddress:0x00007fc909880ed0 @street="49119 Jadon Mills, Apt. 832", @city="Mosciskitown", @state="KS", @postal_code="75290-2015", @country="USA">, @home_phone="1-566-578-7503", @full_mvi_ids=["1000123456V123456^NI^200M^USVHA^P", "12345^PI^516^USVHA^PCE", "2^PI^553^USVHA^PCE", "12345^PI^200HD^USVHA^A", "TKIP123456^PI^200IP^USVHA^A", "123456^PI^200MHV^USVHA^A", "UNK^NI^200DOD^USDOD^A", "12345678^PI^200CORP^USVBA^A"], @icn="27766144085338093", @icn_with_aaid="1000123456V123456^NI^200M^USVHA", @mhv_ids=["12610607526", "65234986041"], @active_mhv_ids=["12610607526", "65234986041"], @vha_facility_ids=[], @edipi="1007697216", @participant_id="600061742", @birls_id="796043735", @sec_id="0001234567", @vet360_id="123456789", @historical_icns=["1000123457V123456", "1000123458V123456"]>, @error=nil>>>, @headers={"va_eauth_csid"=>"DSLogon", "va_eauth_authenticationmethod"=>"DSLogon", "va_eauth_pnidtype"=>"SSN", "va_eauth_assurancelevel"=>"3", "va_eauth_firstName"=>"WESLEY", "va_eauth_lastName"=>"FORD", "va_eauth_issueinstant"=>"2017-12-07T00:55:09Z", "va_eauth_dodedipnid"=>"1007697216", "va_eauth_birlsfilenumber"=>"796043735", "va_eauth_pid"=>"600061742", "va_eauth_pnid"=>"796043735", "va_eauth_birthdate"=>"1986-05-06T00:00:00+00:00", "va_eauth_authorization"=>"{\"authorizationResponse\":{\"status\":\"VETERAN\",\"idType\":\"SSN\",\"id\":\"796043735\",\"edi\":\"1007697216\",\"firstName\":\"WESLEY\",\"lastName\":\"FORD\",\"birthDate\":\"1986-05-06T00:00:00+00:00\"}}", "va_eauth_authenticationauthority"=>"eauth"}> received :perform with unexpected arguments
         expected: (:post, "getCurrentInfo", "", {"ssn"=>"796043735", "Authorization"=>"Token token=PUBLICDEMO123", "Content-Type"=>"application/json"...ondaryPhoneNumberFilled"=>true, "veteranName"=>"JEFF TERRELL WATSON"}, "userPoaInfoAvailable"=>true})
              got: (:post, "getCurrentInfo", nil, {"ssn"=>"796043735", "Authorization"=>"Token token=PUBLICDEMO123", "Content-Type"=>"application/json"...ondaryPhoneNumberFilled"=>true, "veteranName"=>"JEFF TERRELL WATSON"}, "userPoaInfoAvailable"=>true})`

Reply:
I think in that case you should change it to `expect` to receive

Reply:
I'll add an expect here, one sec.

Reply:
Oh looks like @charleystran already added an expect.